pyartifactory/objects.py

```diff
@@ -231,6 +169,91 @@ def unlock(self, name: str) -> None:
+        super().__init__(artifactory)
+        if self._api_version == 2:
+            self._uri = "v2/security/permissions"
+        if self._api_version == 1:
+            self._uri = "security/permissions"
+
+    def create(
+        self, permission: Union[Permission, PermissionV2]
+    ) -> Union[Permission, PermissionV2]:
+        """
+        Creates a permission
+        :param permission: Permission v2 or v1 object
+        :return: Permission v2 or v1
+        """
+        permission_name = permission.name
+        try:
+            self.get(permission_name)
+            logger.debug("Permission %s already exists", permission_name)
+            raise PermissionAlreadyExistsException(
+                f"Permission {permission_name} already exists"
+            )
+        except PermissionNotFoundException:
+            self._put(f"api/{self._uri}/{permission_name}", json=permission.dict())
+            logger.debug("Permission %s successfully created", permission_name)
+            return self.get(permission_name)
+
+    def get(self, permission_name: str) -> Union[Permission, PermissionV2]:
+        """
+        Read permission from artifactory. Fill object if exist
+        :param permission_name: Name of the permission to retrieve
+        :return: Permission
+        """
+        try:
+            response = self._get(f"api/{self._uri}/{permission_name}")
+            logger.debug("Permission %s found", permission_name)
+            return (
+                Permission(**response.json())
+                if self._artifactory.api_version == 1
+                else PermissionV2(**response.json())
+            )
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404 or error.response.status_code == 400:
+                logger.error("Permission %s does not exist", permission_name)
+                raise PermissionNotFoundException(
+                    f"Permission {permission_name} does not exist"
+                )
+            raise ArtifactoryException from error
+
+    def list(self) -> List[SimplePermission]:
+        """
+        Lists all the permissions
+        :return: A list of permissions
+        """
+        response = self._get(f"api/{self._uri}")
+        logger.debug("List all permissions successful")
+        return [SimplePermission(**permission) for permission in response.json()]
+
+    def update(
+        self, permission: Union[Permission, PermissionV2]
+    ) -> Union[Permission, PermissionV2]:
+        """
+        Updates an artifactory permission
+        :param permission: Permission v2 or v1 object
+        :return: Permission v2 or v1
+        """
+        permission_name = permission.name
+        self._put(f"api/{self._uri}/{permission_name}", json=permission.dict())
+        logger.debug("Permission %s successfully updated", permission_name)
+        return self.get(permission_name)
+
+    def delete(self, permission_name: str) -> None:
```

Comment:
I understand, but we made the choice of raising a PermissionNotFoundException when the permission is missing. If we change this behavior here, we would would change it for all delete operations. It would need a new PR especially for this. You're welcome to propose it and I'll be glad to review it.